doc/source/notebooks/intro_to_gpflow2.pct.py

```diff
@@ -495,45 +497,32 @@ def checkpointing_training_loop(
 # %% [markdown]
 # ### TensorFlow `saved_model`
 #
-# At present, TensorFlow does not support saving custom variables like instances of the `gpflow.base.Parameter` class, see [this TensorFlow github issue](https://github.com/tensorflow/tensorflow/issues/34908).
-#
-# However, once training is complete, it is possible to clone the model and replace all `gpflow.base.Parameter`s with `tf.constant`s holding the same value:
-
-# %%
-model
-
-# %%
-frozen_model = gpflow.utilities.freeze(model)
-
-# %% [markdown]
-# In order to save the model we need to define a `tf.Module` holding the `tf.function`'s that we wish to export, as well as a reference to the underlying model:
+# In order to save the model we need to explicitly store the `tf.function`-compiled functions that we wish to export:
 
 # %%
-module_to_save = tf.Module()
-predict_fn = tf.function(
-    frozen_model.predict_f, input_signature=[tf.TensorSpec(shape=[None, 1], dtype=tf.float64)]
+model.predict_f_compiled = tf.function(
+    model.predict_f, input_signature=[tf.TensorSpec(shape=[None, 1], dtype=tf.float64)]
 )
-module_to_save.predict = predict_fn
 
 # %% [markdown]
-# Save original result for futher comparison. We also convert `samples_input` to a tensor. For a tensor input a `tf.function` will compile a single graph.
+# We also save the original prediction for later comparison. Here, `samples_input` needs to be a tensor, so that `tf.function` will compile a single graph:
```

Comment:
```suggestion
# We also save the original prediction for later comparison. Here `samples_input` needs to be a tensor so that `tf.function` will compile a single graph:
```